### PR TITLE
add property controller, dao, and integration tests

### PR DESCRIPTION
**Pull request by adamkorynta**
Adds features to handle properties in the Content Data API (CDA). It adds a new PropertyController and PropertyDao to manage access and manipulation of properties in the database layer.

Java client: https://github.com/HydrologicEngineeringCenter/cwms-data-api-client/pull/167

### CODE REVIEW

File: cwms-data-api/src/test/java/cwms/cda/api/PropertyControllerIT.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        TestAccounts.KeyUser user = TestAccounts.KeyUser.SPK_NORMAL;
+
+        //Create the property
+        given()
+                .log().ifValidationFails(LogDetail.ALL, true)
+                .accept(Formats.JSONV2)
+                .contentType(Formats.JSONV2)
+                .body(json)
+                .header(AUTH_HEADER, user.toHeaderValue())
+                .when()
```
Comment (MikeNeilson) [inline cwms-data-api/src/test/java/cwms/cda/api/PropertyControllerIT.java]: unindent .when and .then. (also looks like it's all indented twice.)

**Review by MikeNeilson — CHANGES_REQUESTED** [review]
Some formatting got away from you, but otherwise looks great.

File: cwms-data-api/src/main/java/cwms/cda/api/PropertyController.java [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+            retVal = om.readValue(body, Property.class);
+        } catch (Exception e) {
+            throw new IOException("Failed to deserialize property", e);
+        }
+        return retVal;
+    }
+
+    private static ObjectMapper getObjectMapperForFormat(ContentType contentType) {
+        ObjectMapper om;
+        if (ContentType.equivalent(Formats.JSONV2, contentType.toString())) {
```
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/api/PropertyController.java]: The content-type mapper was design so you can just pass the data type in and with an annotation on the datatype get the right formatter given the content-type string.

This function is duplication of all that logic.

**Review by MikeNeilson — CHANGES_REQUESTED** [review]

File: cwms-data-api/src/main/java/cwms/cda/api/PropertyController.java (hunk 2 above)
Comment (adamkorynta) [reply cwms-data-api/src/main/java/cwms/cda/api/PropertyController.java]: Looks like that was only done one-way. I implemented the deserialization and updated my code to use the Formats factory methods. I'll do a follow-up PR removing the redundant code that I had copied this logic from.

File: cwms-data-api/src/test/java/cwms/cda/api/PropertyControllerIT.java (hunk 1 above)
Comment (adamkorynta) [reply cwms-data-api/src/test/java/cwms/cda/api/PropertyControllerIT.java]: oops, missed one, updated

**Review by MikeNeilson — APPROVED** [review]
Nice, thanks for the additional improvement for parsing the content as well.

**Pull request merged**